src/aatoolbox/datasources/usgs/usgs_ndvi.py

```diff
@@ -0,0 +1,851 @@
+difference from the previous year, and median anomaly
+presented as a percentile.
+
+Data by USGS is published quickly after the dekad.
+After about 1 month this data is updated with temporal smoothing
+and error correction for cloud cover. Files for a specific
+dekad and region can range from 30MB up to over 100MB, so
+downloading and processing can take a long time.
+"""
+
+# TODO: add progress bar
+import functools
+import logging
+import re
+from datetime import date, datetime
+from io import BytesIO
+from pathlib import Path
+from typing import Generator, List, Optional, Tuple, Union
+from urllib.error import HTTPError
+from urllib.request import urlopen
+from zipfile import ZipFile
+
+import geopandas as gpd
+import pandas as pd
+import rioxarray  # noqa: F401
+import xarray as xr
+from rasterio.errors import RasterioIOError
+
+import aatoolbox.utils.raster  # noqa: F401
+from aatoolbox.config.countryconfig import CountryConfig
+from aatoolbox.datasources.datasource import DataSource
+from aatoolbox.utils.dates import (
+    _compare_dekads_gt,
+    _compare_dekads_lt,
+    _dekad_to_date,
+    _expand_dekads,
+    _get_dekadal_date,
+)
+
+logger = logging.getLogger(__name__)
+
+_DATE_TYPE = Union[date, str, Tuple[int, int], None]
+_EARLIEST_DATE = (2002, 19)
+
+
+class _UsgsNdvi(DataSource):
+    """Base class to retrieve USGS NDVI data.
+
+    Parameters
+    ----------
+    country_config : CountryConfig
+        Country configuration
+    data_variable : str
+        Data variable date
+    data_variable_suffix : str
+        Data variable file string
+    data_variable_url : str
+        URL string for data variable
+    start_date : _DATE_TYPE, default = None
+        Start date. Can be passed as a ``datetime.date``
+        object or a data string in ISO8601 format, and
+        the relevant dekad will be determined. Or pass
+        directly as year-dekad tuple, e.g. (2020, 1).
+        If ``None``, ``start_date`` is set to earliest
+        date with data: 2002, dekad 19.
+    end_date : _DATE_TYPE, default = None
+        End date. Can be passed as a ``datetime.date``
+        object and the relevant dekad will be determined,
+        as a date string in ISO8601 format, or as a
+        year-dekad tuple, i.e. (2020, 1). If ``None``,
+        ``end_date`` is set to ``date.today()``.
+    """
+
+    def __init__(
+        self,
+        country_config: CountryConfig,
+        data_variable: str,
+        data_variable_suffix: str,
+        data_variable_url: str,
+        start_date: _DATE_TYPE = None,
+        end_date: _DATE_TYPE = None,
+    ):
+        super().__init__(
+            country_config=country_config,
+            datasource_base_dir="usgs_ndvi",
+            is_public=True,
+            is_global_raw=True,
+            config_datasource_name="usgs_ndvi",
+        )
+
+        # set area url and prefix from config
+        if self._country_config.usgs_ndvi is None:
```

Comment:
This shouldn't be needed as it's taken care of when you pass `config_datasource_name` to the parent

Reply:
That's what I thought, but it was still in `fewsnet.py` so was wondering maybe it is required for some reason.

Reply:
Oops that should be deleted, maybe you can sneak it in here?

Reply:
Yeah can do!

Reply:
Would you mind pointing me to the code where this validation is done? Interested to see how you made the setup work, it's cool code but not entirely clear on how it's working ha!

Reply:
I guess it's done in the `_config_attribute_name_validator`.

Reply:
Yeah exactly!

Reply:
This isn't actually raising an `AttributeError` in the FEWS NET test because `getattr()` doesn't return an error when it's `None`, the default.

Reply:
Awesome, thanks for taking care of this for FN as well!